core/org.eclipse.cdt.ui/src/org/eclipse/cdt/ui/CElementSorter.java

```diff
@@ -427,20 +432,28 @@ private ISourceRoot getSourceRoot(Object element) {
 		return (ISourceRoot) celement;
 	}
 
-	private int compareWithLabelProvider(Viewer viewer, Object e1, Object e2) {
+	// implementation taken from org.eclipse.jdt.ui.JavaElementComparator
+	private String getNonCElementLabel(Viewer viewer, Object element) {
+		// try to use the workbench adapter for non - C++ resources or if not available, use the viewers label provider
+		if (element instanceof IResource) {
+			return ((IResource) element).getName();
+		}
+		if (element instanceof IStorage) {
+			return ((IStorage) element).getName();
+		}
+		if (element instanceof IAdaptable) {
```

Comment:
Here it is better to use `Adapters#adapt`, because a lot of elements adapted "externally", with `IAdapterFactory`

Reply:
I prefer consistency with JDT if possible. See also https://github.com/eclipse-cdt/cdt/issues/716#issuecomment-1961552164

Reply:
> Please let me know if matching JDT is a problem here.

The problem _may_ appear for element contributed by 3rd party, but currently I'm not aware of any.

Reply:
Anyway, this was a generic recommendation regarding Platform usage, should not be considered as an obstacle to merge this one.

Reply:
+1 I agree - new code should definitely use Adapters#adapt.